

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>BatMeth2: Alignment &mdash; BatMeth2 1.1 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home" alt="Documentation Home"> BatMeth2
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">BatMeth2: Alignment</a></li>
</ul>
</div>
            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">BatMeth2</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>BatMeth2: Alignment</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/Alignment.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="batmeth2-alignment">
<h1>BatMeth2: Alignment<a class="headerlink" href="#batmeth2-alignment" title="Permalink to this headline">¶</a></h1>
<p><strong>Program: BatMeth2  align</strong></p>
<p>Version: v1.0</p>
<p>[ Single-end-reads ]</p>
<p><strong>Usage:</strong>   batmeth2 -g INDEX -i INPUT -o OUTPUT</p>
<p>Example:   batmeth2 -g /data/index/hg19/hg19.fa -i Read.fq -o outPrefix -p 10</p>
<p>[ Paired-end-reads ]</p>
<p><strong>Usage:</strong>   batmeth2 -g INDEX -i INPUT_left -i INPUT_right -o OUTPUT</p>
<p>Example:   batmeth2 -g /data/index/hg19/hg19.fa -i Read_R1_left.fq -i Read_R2_right.fq -o outPrefix -p 10</p>
<p>Parameters :</p>
<p>​      –inputfile | -i &lt;filename&gt;   Name of input file</p>
<p>​      –genome | -g &lt;filename&gt;      Name of the genome mapped against</p>
<p>​      –outputfile | -o &lt;filename&gt;  Name of output file prefix</p>
<p>​      –indelsize                   indel size</p>
<p>​      –non_directional             Alignments to all four bisulfite strands will be reported. Default: OFF.</p>
<p>​      –insertsize | -s             inital insert size, default 800</p>
<p>​      –std | -d &lt;integer&gt;          standard deviatiion of reads distribution</p>
<p>​      –flanksize | -f &lt;integer&gt;    size of flanking region for Smith-Waterman</p>
<p>​      –swlimit | &lt;integer&gt;         try at most &lt;integer&gt; sw extensions</p>
<p>​      –threads | -p &lt;integer&gt;      Launch &lt;integer&gt; threads</p>
<p>​      –NoInDels | -I               not to find the indels result</p>
<p>​      –help | -h                   Print help</p>
<p>Note: To use BatMeth2, you need to first index the genome with <a href="#id1"><span class="problematic" id="id2">`</span></a>build_all genome’.</p>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, momocoding

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>