
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Calculate DNA methylation level &#8212; BatMeth2 1.1 documentation</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=62ba249389abaaa9ffc34bf36a076bdc1d65ee18" type="text/css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=f31d14ad54b65d19161ba51d4ffff3a77ae00456"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="bmtools" href="bmtools.html" />
    <link rel="prev" title="Alignment" href="Alignment.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
      
      
      <h1 class="site-logo" id="site-title">BatMeth2 1.1 documentation</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   Installation
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="example.html">
   Example Data
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bdindex.html">
   Build index
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Pipeline.html">
   Pipeline
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Alignment.html">
   Alignment
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Calculate DNA methylation level
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="bmtools.html">
   bmtools
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="PlotMeth.html">
   PlotMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="DiffMeth.html">
   DiffMeth
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="Requirements.html">
   Requirements
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/function/Calmeth.rst.txt"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.rst</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#calmeth">
   Calmeth
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#paramaters">
   Paramaters
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-files">
   Output files
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-file-format">
   Output file format
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Calculate DNA methylation level</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#calmeth">
   Calmeth
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#paramaters">
   Paramaters
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-files">
   Output files
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#output-file-format">
   Output file format
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section id="calculate-dna-methylation-level">
<h1>Calculate DNA methylation level<a class="headerlink" href="#calculate-dna-methylation-level" title="Permalink to this headline">#</a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#calmeth" id="id1">Calmeth</a></p></li>
<li><p><a class="reference internal" href="#paramaters" id="id2">Paramaters</a></p></li>
<li><p><a class="reference internal" href="#output-files" id="id3">Output files</a></p></li>
<li><p><a class="reference internal" href="#output-file-format" id="id4">Output file format</a></p></li>
</ul>
</div>
<section id="calmeth">
<h2><a class="toc-backref" href="#id1">Calmeth</a><a class="headerlink" href="#calmeth" title="Permalink to this headline">#</a></h2>
<p>Calculate DNA methylation level from alignment files, you can obtained single-base cytosine DNA
methylation results, and the chromosome region DNA methylation levels files.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
  with bam file:
    calmeth <span class="o">[</span>options<span class="o">]</span> -g genome.fa  -b alignment.sort.bam -m output.methrario.mbw
  with sam file:
    calmeth <span class="o">[</span>options<span class="o">]</span> -g genome.fa  -i alignment.sort.sam -m output.methrario.mbw
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The bam or sam file MUST sorted by <cite>samtools sort</cite>.</p>
</div>
</section>
<section id="paramaters">
<h2><a class="toc-backref" href="#id2">Paramaters</a><a class="headerlink" href="#paramaters" title="Permalink to this headline">#</a></h2>
<table class="table">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" colspan="3"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>-m/–methratio</p></td>
<td><p>[MethFileNamePrefix]  Predix of methratio output file (mbw)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–genome/-g</p></td>
<td><p>Name of the genome mapped against, MUST build index first <a class="reference internal" href="bdindex.html"><span class="doc">Build index</span></a></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-i/–input</p></td>
<td><p>Sam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b/–binput</p></td>
<td><p>Bam format file, sorted by samtools sort.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-Q [int]</p></td>
<td><p>caculate the methratio while read QulityScore &gt;= Q. default:20</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-n [float]</p></td>
<td><p>Number of mismatches, default 0.06 percentage of read length. [0-1]</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-c|–coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default:4</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-nC</p></td>
<td><p>&gt;= &lt;INT&gt; Cs per region. default:1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-R/–Regions</p></td>
<td><p>Bins for DMR caculate , default 1000(1kb) .</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>–binsfile</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>DNA methylation level distributions in chrosome, default output file: {Prefix}.methBins.txt</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>-s/–step</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Chrosome using an overlapping sliding window of 100000bp at a step of 50000bp.
default step: 50000(bp)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-r/–remove_dup</p></td>
<td><p>REMOVE_DUP, default:true</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-f|–sam [outfile]</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>f for sam format outfile contain methState.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–sam-seq-beforeBS</p></td>
<td><p>Converting BS read to the genome sequences.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–help/-h</p></td>
<td><p>Print help</p></td>
</tr>
</tbody>
</table>
</section>
<section id="output-files">
<h2><a class="toc-backref" href="#id3">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this headline">#</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. prefix.methratio.mbw
<span class="m">2</span>. prefix.methBins.txt
<span class="m">3</span>. prefix_Region.CG/CHG/CHH.txt
<span class="m">4</span>. prefix.mCdensity.txt
<span class="m">5</span>. prefix.mCcatero.txt
</pre></div>
</div>
</section>
<section id="output-file-format">
<h2><a class="toc-backref" href="#id4">Output file format</a><a class="headerlink" href="#output-file-format" title="Permalink to this headline">#</a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. methratio.mbw <span class="o">(</span>binary file, view and processed with bmtools <span class="k">in</span> BatMeth2<span class="o">)</span>
    Chromosome Loci <span class="o">(</span>end<span class="o">)</span> methlevel CT_count Strand Context
    <span class="c1"># ex. bmtools view -i test.mr.mbw -r Chr1:61-61</span>
    <span class="c1"># ex. Chr1    61      0.286364  11  +       CHH</span>
    <span class="c1"># CT_count     The number of coverage in this base pair.</span>
    <span class="c1"># BigWig for genome browser, suggest with end col.</span>
<span class="m">2</span>. methBins
    Chrom BinIndex methlevel context
    <span class="c1"># ex. Chr1    1       0.113674        CG</span>
    <span class="c1"># The BinIndex is defined by -s paramater in calmeth.</span>
    <span class="c1"># This file can be used for visualization the DNA methylation level acorss the chromosome.</span>
<span class="m">3</span>. Region
    chrom regionStart strand context c_count ct_count
    <span class="c1"># ex. Chr1    1001    +       CG      1       227</span>
    <span class="c1"># The bins methylation level output file (BS.mr_Region.C*.txt) can be used to do DMR detection.</span>
<span class="m">4</span>. mCdensity
    CG/CHG/CHH C count <span class="k">in</span> <span class="o">[</span><span class="m">0</span>, <span class="m">1</span>%<span class="o">)</span> <span class="o">[</span><span class="m">1</span>%, <span class="m">2</span>%<span class="o">)</span> ... <span class="o">[</span><span class="m">49</span>%, <span class="m">50</span>%<span class="o">)</span> ... <span class="o">[</span><span class="m">99</span>%, <span class="m">100</span>%<span class="o">]</span>
    <span class="c1"># According to the DNA methylation level, the number of cytosine sites at different methylation levels was counted from 0 to 100.</span>
<span class="m">5</span>. mCcatero
    Average DNA methylation level including mC, mCG and other states.
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BatMeth2">on github</a>.</p>
</div>
</section>
</section>


              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="Alignment.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Alignment</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="bmtools.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">bmtools</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By momocoding<br/>
  
      &copy; Copyright 2020, momocoding.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>