

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Function &mdash; BatMeth2 1.1 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Build index" href="bdindex.html" />
    <link rel="prev" title="Installation" href="install.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> BatMeth2
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Function</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#general-principles">General principles</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#parameters-to-decrease-the-run-time">Parameters to decrease the run time</a></li>
<li class="toctree-l3"><a class="reference internal" href="#filtering-bams-while-processing">Filtering BAMs while processing</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#tools-for-bam-and-bigwig-file-processing">Tools for BAM and bigWig file processing</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tools-multibamsummary"><span class="xref std std-doc">tools/multiBamSummary</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-multibigwigsummary"><span class="xref std std-doc">tools/multiBigwigSummary</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-correctgcbias"><span class="xref std std-doc">tools/correctGCBias</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-bamcoverage"><span class="xref std std-doc">tools/bamCoverage</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-bamcompare"><span class="xref std std-doc">tools/bamCompare</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-bigwigcompare"><span class="xref std std-doc">tools/bigwigCompare</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-computematrix"><span class="xref std std-doc">tools/computeMatrix</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-alignmentsieve"><span class="xref std std-doc">tools/alignmentSieve</span></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#tools-for-qc">Tools for QC</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotcorrelation"><span class="xref std std-doc">tools/plotCorrelation</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotpca"><span class="xref std std-doc">tools/plotPCA</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotfingerprint"><span class="xref std std-doc">tools/plotFingerprint</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-bampefragmentsize"><span class="xref std std-doc">tools/bamPEFragmentSize</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-computegcbias"><span class="xref std std-doc">tools/computeGCBias</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotcoverage"><span class="xref std std-doc">tools/plotCoverage</span></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#heatmaps-and-summary-plots">Heatmaps and summary plots</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotheatmap"><span class="xref std std-doc">tools/plotHeatmap</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotprofile"><span class="xref std std-doc">tools/plotProfile</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-plotenrichment"><span class="xref std std-doc">tools/plotEnrichment</span></a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#miscellaneous">Miscellaneous</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tools-computematrixoperations"><span class="xref std std-doc">tools/computeMatrixOperations</span></a></li>
<li class="toctree-l3"><a class="reference internal" href="#tools-estimatereadfiltering"><span class="xref std std-doc">tools/estimateReadFiltering</span></a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="bdindex.html">Build index</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">BatMeth2</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Function</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/function/function.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="function">
<h1>Function<a class="headerlink" href="#function" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>BatMeth2 可以完成流程分析以及分步骤完成以下模块:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>Alignment</strong> <span class="xref std std-doc">function/Alignment</span></p></li>
<li><p><strong>Calmeth</strong> <span class="xref std std-doc">function/Calmeth</span></p></li>
<li><p><strong>MethyGff</strong> <span class="xref std std-doc">function/MethyGff</span></p></li>
<li><p><strong>bt2bigwig</strong> <span class="xref std std-doc">function/bt2bigwig</span></p></li>
<li><p><strong>PlotMeth</strong> <span class="xref std std-doc">function/PlotMeth</span>.</p></li>
<li><p><strong>DiffMeth</strong> <span class="xref std std-doc">function/DiffMeth</span>.</p></li>
</ul>
</div></blockquote>
<p>For more, see <span class="xref std std-doc">changelog</span>.</p>
</div>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#general-principles" id="id1">General principles</a></p>
<ul>
<li><p><a class="reference internal" href="#parameters-to-decrease-the-run-time" id="id2">Parameters to decrease the run time</a></p></li>
<li><p><a class="reference internal" href="#filtering-bams-while-processing" id="id3">Filtering BAMs while processing</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#tools-for-bam-and-bigwig-file-processing" id="id4">Tools for BAM and bigWig file processing</a></p>
<ul>
<li><p><a class="reference internal" href="#tools-multibamsummary" id="id5"><span class="xref std std-doc">tools/multiBamSummary</span></a></p></li>
<li><p><a class="reference internal" href="#tools-multibigwigsummary" id="id6"><span class="xref std std-doc">tools/multiBigwigSummary</span></a></p></li>
<li><p><a class="reference internal" href="#tools-correctgcbias" id="id7"><span class="xref std std-doc">tools/correctGCBias</span></a></p></li>
<li><p><a class="reference internal" href="#tools-bamcoverage" id="id8"><span class="xref std std-doc">tools/bamCoverage</span></a></p></li>
<li><p><a class="reference internal" href="#tools-bamcompare" id="id9"><span class="xref std std-doc">tools/bamCompare</span></a></p></li>
<li><p><a class="reference internal" href="#tools-bigwigcompare" id="id10"><span class="xref std std-doc">tools/bigwigCompare</span></a></p></li>
<li><p><a class="reference internal" href="#tools-computematrix" id="id11"><span class="xref std std-doc">tools/computeMatrix</span></a></p></li>
<li><p><a class="reference internal" href="#tools-alignmentsieve" id="id12"><span class="xref std std-doc">tools/alignmentSieve</span></a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#tools-for-qc" id="id13">Tools for QC</a></p>
<ul>
<li><p><a class="reference internal" href="#tools-plotcorrelation" id="id14"><span class="xref std std-doc">tools/plotCorrelation</span></a></p></li>
<li><p><a class="reference internal" href="#tools-plotpca" id="id15"><span class="xref std std-doc">tools/plotPCA</span></a></p></li>
<li><p><a class="reference internal" href="#tools-plotfingerprint" id="id16"><span class="xref std std-doc">tools/plotFingerprint</span></a></p></li>
<li><p><a class="reference internal" href="#tools-bampefragmentsize" id="id17"><span class="xref std std-doc">tools/bamPEFragmentSize</span></a></p></li>
<li><p><a class="reference internal" href="#tools-computegcbias" id="id18"><span class="xref std std-doc">tools/computeGCBias</span></a></p></li>
<li><p><a class="reference internal" href="#tools-plotcoverage" id="id19"><span class="xref std std-doc">tools/plotCoverage</span></a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#heatmaps-and-summary-plots" id="id20">Heatmaps and summary plots</a></p>
<ul>
<li><p><a class="reference internal" href="#tools-plotheatmap" id="id21"><span class="xref std std-doc">tools/plotHeatmap</span></a></p></li>
<li><p><a class="reference internal" href="#tools-plotprofile" id="id22"><span class="xref std std-doc">tools/plotProfile</span></a></p></li>
<li><p><a class="reference internal" href="#tools-plotenrichment" id="id23"><span class="xref std std-doc">tools/plotEnrichment</span></a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#miscellaneous" id="id24">Miscellaneous</a></p>
<ul>
<li><p><a class="reference internal" href="#tools-computematrixoperations" id="id25"><span class="xref std std-doc">tools/computeMatrixOperations</span></a></p></li>
<li><p><a class="reference internal" href="#tools-estimatereadfiltering" id="id26"><span class="xref std std-doc">tools/estimateReadFiltering</span></a></p></li>
</ul>
</li>
</ul>
</div>
<table class="docutils align-default">
<colgroup>
<col style="width: 17%" />
<col style="width: 8%" />
<col style="width: 17%" />
<col style="width: 20%" />
<col style="width: 38%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>tool</p></th>
<th class="head"><p>type</p></th>
<th class="head"><p>input files</p></th>
<th class="head"><p>main output file(s)</p></th>
<th class="head"><p>application</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/multiBamSummary</span></p></td>
<td><p>data integration</p></td>
<td><p>2 or more BAM</p></td>
<td><p>interval-based table of values</p></td>
<td><p>perform cross-sample analyses of read counts –&gt; plotCorrelation, plotPCA</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/multiBigwigSummary</span></p></td>
<td><p>data integration</p></td>
<td><p>2 or more bigWig</p></td>
<td><p>interval-based table of values</p></td>
<td><p>perform cross-sample analyses of genome-wide scores –&gt; plotCorrelation, plotPCA</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/plotCorrelation</span></p></td>
<td><p>visualization</p></td>
<td><p>bam/multiBigwigSummary output</p></td>
<td><p>clustered heatmap</p></td>
<td><p>visualize the Pearson/Spearman correlation</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/plotPCA</span></p></td>
<td><p>visualization</p></td>
<td><p>bam/multiBigwigSummary output</p></td>
<td><p>2 PCA plots</p></td>
<td><p>visualize the principal component analysis</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/plotFingerprint</span></p></td>
<td><p>QC</p></td>
<td><p>2 BAM</p></td>
<td><p>1 diagnostic plot</p></td>
<td><p>assess enrichment strength of a ChIP sample</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/computeGCBias</span></p></td>
<td><p>QC</p></td>
<td><p>1 BAM</p></td>
<td><p>2 diagnostic plots</p></td>
<td><p>calculate the exp. and obs. GC distribution of reads</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/correctGCBias</span></p></td>
<td><p>QC</p></td>
<td><p>1 BAM, output from computeGCbias</p></td>
<td><p>1 GC-corrected BAM</p></td>
<td><p>obtain a BAM file with reads distributed according to the genome’s GC content</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/bamCoverage</span></p></td>
<td><p>normalization</p></td>
<td><p>BAM</p></td>
<td><p>bedGraph or bigWig</p></td>
<td><p>obtain the normalized read coverage of a single BAM file</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/bamCompare</span></p></td>
<td><p>normalization</p></td>
<td><p>2 BAM</p></td>
<td><p>bedGraph or bigWig</p></td>
<td><p>normalize 2 files to each other (e.g. log2ratio, difference)</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/computeMatrix</span></p></td>
<td><p>data integration</p></td>
<td><p>1 or more bigWig, 1 or more BED</p></td>
<td><p>zipped file for plotHeatmap or plotProfile</p></td>
<td><p>compute the values needed for heatmaps and summary plots</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/estimateReadFiltering</span></p></td>
<td><p>information</p></td>
<td><p>1 or more BAM files</p></td>
<td><p>table of values</p></td>
<td><p>estimate the number of reads filtered from a BAM file or files</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/alignmentSieve</span></p></td>
<td><p>QC</p></td>
<td><p>1 BAM file</p></td>
<td><p>1 filtered BAM or BEDPE file</p></td>
<td><p>filters a BAM file based on one or more criteria</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/plotHeatmap</span></p></td>
<td><p>visualization</p></td>
<td><p>computeMatrix output</p></td>
<td><p>heatmap of read coverages</p></td>
<td><p>visualize the read coverages for genomic regions</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/plotProfile</span></p></td>
<td><p>visualization</p></td>
<td><p>computeMatrix output</p></td>
<td><p>summary plot (“meta-profile”)</p></td>
<td><p>visualize the average read coverages over a group of genomic regions</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/plotCoverage</span></p></td>
<td><p>visualization</p></td>
<td><p>1 or more BAM</p></td>
<td><p>2 diagnostic plots</p></td>
<td><p>visualize the average read coverages over sampled genomic  positions</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/bamPEFragmentSize</span></p></td>
<td><p>information</p></td>
<td><p>1  BAM</p></td>
<td><p>text with paired-end fragment length</p></td>
<td><p>obtain the average fragment length from paired ends</p></td>
</tr>
<tr class="row-even"><td><p><span class="xref std std-doc">tools/plotEnrichment</span></p></td>
<td><p>visualization</p></td>
<td><p>1 or more BAM and 1 or more BED/GTF</p></td>
<td><p>A diagnostic plot</p></td>
<td><p>plots the fraction of alignments overlapping the given features</p></td>
</tr>
<tr class="row-odd"><td><p><span class="xref std std-doc">tools/computeMatrixOperations</span></p></td>
<td><p>miscellaneous</p></td>
<td><p>1 or more BAM and 1 or more BED/GTF</p></td>
<td><p>A diagnostic plot</p></td>
<td><p>plots the fraction of alignments overlapping the given features</p></td>
</tr>
</tbody>
</table>
<div class="section" id="general-principles">
<h2><a class="toc-backref" href="#id1">General principles</a><a class="headerlink" href="#general-principles" title="Permalink to this headline">¶</a></h2>
<p>A typical deepTools command could look like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bamCoverage --bam myAlignedReads.bam <span class="se">\</span>
--outFileName myCoverageFile.bigWig <span class="se">\</span>
--outFileFormat bigwig <span class="se">\</span>
--fragmentLength <span class="m">200</span> <span class="se">\</span>
--ignoreDuplicates <span class="se">\</span>
--scaleFactor <span class="m">0</span>.5
</pre></div>
</div>
<p>You can always see all available command-line options via –help:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bamCoverage --help
</pre></div>
</div>
<ul class="simple">
<li><p>Output format of plots should be indicated by the file ending, e.g. <code class="docutils literal notranslate"><span class="pre">MyPlot.pdf</span></code> will return a pdf file, <code class="docutils literal notranslate"><span class="pre">MyPlot.png</span></code> a png-file</p></li>
<li><p>All tools that produce plots can also output the underlying data - this can be useful in cases where you don’t like the deepTools visualization, as you can then use the data matrices produced by deepTools with your favorite plotting tool, such as R</p></li>
<li><p>The vast majority of command line options are also available in Galaxy (in a few cases with minor changes to their naming).</p></li>
</ul>
<div class="section" id="parameters-to-decrease-the-run-time">
<h3><a class="toc-backref" href="#id2">Parameters to decrease the run time</a><a class="headerlink" href="#parameters-to-decrease-the-run-time" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">numberOfProcessors</span></code> - Number of processors to be used</dt><dd><dl class="simple">
<dt>For example, setting <code class="docutils literal notranslate"><span class="pre">--numberOfProcessors</span> <span class="pre">10</span></code> will split up the</dt><dd><p>workload internally into 10 chunks, which will be
processed in parallel.</p>
</dd>
</dl>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">region</span></code> - Process only a single genomic region.</dt><dd><p>This is particularly useful when you’re still trying    to figure out the best parameter setting. You can focus on a certain genomic region by setting, e.g., <code class="docutils literal notranslate"><span class="pre">--region</span> <span class="pre">chr2</span></code> or
<code class="docutils literal notranslate"><span class="pre">--region</span> <span class="pre">chr2:100000-200000</span></code></p>
</dd>
</dl>
</li>
</ul>
<p>These parameters are optional and available throughout almost all deepTools.</p>
</div>
<div class="section" id="filtering-bams-while-processing">
<h3><a class="toc-backref" href="#id3">Filtering BAMs while processing</a><a class="headerlink" href="#filtering-bams-while-processing" title="Permalink to this headline">¶</a></h3>
<p>Several deepTools modules allow for efficient processing of BAM files, e.g. <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> and <code class="docutils literal notranslate"><span class="pre">bamCompare</span></code>.
We offer several ways to filter those BAM files on the fly so that you don’t need to pre-process them using other tools such as <a class="reference external" href="http://www.htslib.org/">samtools</a></p>
<ul class="simple">
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">ignoreDuplicates</span></code></dt><dd><p>Reads with the same orientation and start position will be considered only once. If reads are paired, the mate is also evaluated</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">minMappingQuality</span></code></dt><dd><p>Only reads with a mapping quality score of at least this are considered</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">samFlagInclude</span></code></dt><dd><p>Include reads based on the SAM flag, e.g. <code class="docutils literal notranslate"><span class="pre">--samFlagInclude</span> <span class="pre">64</span></code> gets reads that are first in a pair. For translating SAM flags into English, go to: <a class="reference external" href="https://broadinstitute.github.io/picard/explain-flags.html">https://broadinstitute.github.io/picard/explain-flags.html</a></p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt><code class="docutils literal notranslate"><span class="pre">samFlagExclude</span></code></dt><dd><p>Exclude reads based on the SAM flags - see previous explanation.</p>
</dd>
</dl>
</li>
</ul>
<p>These parameters are optional and available throughout deepTools.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In version 2.3 we introduced a sampling method to correct the effect of filtering when normalizing using <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> or <code class="docutils literal notranslate"><span class="pre">bamCompare</span></code>. For previous versions, if you know that your files will be strongly affected by  the filtering  of duplicates or reads of low quality then consider removing  those reads <em>before</em> using <code class="docutils literal notranslate"><span class="pre">bamCoverage</span></code> or <code class="docutils literal notranslate"><span class="pre">bamCompare</span></code>, as the filtering  by deepTools is done <em>after</em> the scaling factors are calculated!</p>
</div>
</div>
</div>
<div class="section" id="tools-for-bam-and-bigwig-file-processing">
<h2><a class="toc-backref" href="#id4">Tools for BAM and bigWig file processing</a><a class="headerlink" href="#tools-for-bam-and-bigwig-file-processing" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tools-multibamsummary">
<h3><a class="toc-backref" href="#id5"><span class="xref std std-doc">tools/multiBamSummary</span></a><a class="headerlink" href="#tools-multibamsummary" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-multibigwigsummary">
<h3><a class="toc-backref" href="#id6"><span class="xref std std-doc">tools/multiBigwigSummary</span></a><a class="headerlink" href="#tools-multibigwigsummary" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-correctgcbias">
<h3><a class="toc-backref" href="#id7"><span class="xref std std-doc">tools/correctGCBias</span></a><a class="headerlink" href="#tools-correctgcbias" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-bamcoverage">
<h3><a class="toc-backref" href="#id8"><span class="xref std std-doc">tools/bamCoverage</span></a><a class="headerlink" href="#tools-bamcoverage" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-bamcompare">
<h3><a class="toc-backref" href="#id9"><span class="xref std std-doc">tools/bamCompare</span></a><a class="headerlink" href="#tools-bamcompare" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-bigwigcompare">
<h3><a class="toc-backref" href="#id10"><span class="xref std std-doc">tools/bigwigCompare</span></a><a class="headerlink" href="#tools-bigwigcompare" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-computematrix">
<h3><a class="toc-backref" href="#id11"><span class="xref std std-doc">tools/computeMatrix</span></a><a class="headerlink" href="#tools-computematrix" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-alignmentsieve">
<h3><a class="toc-backref" href="#id12"><span class="xref std std-doc">tools/alignmentSieve</span></a><a class="headerlink" href="#tools-alignmentsieve" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="tools-for-qc">
<h2><a class="toc-backref" href="#id13">Tools for QC</a><a class="headerlink" href="#tools-for-qc" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tools-plotcorrelation">
<h3><a class="toc-backref" href="#id14"><span class="xref std std-doc">tools/plotCorrelation</span></a><a class="headerlink" href="#tools-plotcorrelation" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-plotpca">
<h3><a class="toc-backref" href="#id15"><span class="xref std std-doc">tools/plotPCA</span></a><a class="headerlink" href="#tools-plotpca" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-plotfingerprint">
<h3><a class="toc-backref" href="#id16"><span class="xref std std-doc">tools/plotFingerprint</span></a><a class="headerlink" href="#tools-plotfingerprint" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-bampefragmentsize">
<h3><a class="toc-backref" href="#id17"><span class="xref std std-doc">tools/bamPEFragmentSize</span></a><a class="headerlink" href="#tools-bampefragmentsize" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-computegcbias">
<h3><a class="toc-backref" href="#id18"><span class="xref std std-doc">tools/computeGCBias</span></a><a class="headerlink" href="#tools-computegcbias" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-plotcoverage">
<h3><a class="toc-backref" href="#id19"><span class="xref std std-doc">tools/plotCoverage</span></a><a class="headerlink" href="#tools-plotcoverage" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="heatmaps-and-summary-plots">
<h2><a class="toc-backref" href="#id20">Heatmaps and summary plots</a><a class="headerlink" href="#heatmaps-and-summary-plots" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tools-plotheatmap">
<h3><a class="toc-backref" href="#id21"><span class="xref std std-doc">tools/plotHeatmap</span></a><a class="headerlink" href="#tools-plotheatmap" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-plotprofile">
<h3><a class="toc-backref" href="#id22"><span class="xref std std-doc">tools/plotProfile</span></a><a class="headerlink" href="#tools-plotprofile" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-plotenrichment">
<h3><a class="toc-backref" href="#id23"><span class="xref std std-doc">tools/plotEnrichment</span></a><a class="headerlink" href="#tools-plotenrichment" title="Permalink to this headline">¶</a></h3>
</div>
</div>
<div class="section" id="miscellaneous">
<h2><a class="toc-backref" href="#id24">Miscellaneous</a><a class="headerlink" href="#miscellaneous" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tools-computematrixoperations">
<h3><a class="toc-backref" href="#id25"><span class="xref std std-doc">tools/computeMatrixOperations</span></a><a class="headerlink" href="#tools-computematrixoperations" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="tools-estimatereadfiltering">
<h3><a class="toc-backref" href="#id26"><span class="xref std std-doc">tools/estimateReadFiltering</span></a><a class="headerlink" href="#tools-estimatereadfiltering" title="Permalink to this headline">¶</a></h3>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="bdindex.html" class="btn btn-neutral float-right" title="Build index" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="install.html" class="btn btn-neutral float-left" title="Installation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, momocoding

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>