<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Calulate mC across predefined regions &mdash; BatMeth2 1.1 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="../index.html" class="icon icon-home"> BatMeth2
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Example Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="bdindex.html">Build index</a></li>
<li class="toctree-l1"><a class="reference internal" href="Pipeline.html">Pipeline</a></li>
<li class="toctree-l1"><a class="reference internal" href="Alignment.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="Calmeth.html">Calculate DNA methylation level</a></li>
<li class="toctree-l1"><a class="reference internal" href="bmtools.html">bmtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="PlotMeth.html">PlotMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">DiffMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">BatMeth2</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
      <li>Calulate mC across predefined regions</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/function/MethyGff.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="calulate-mc-across-predefined-regions">
<h1>Calulate mC across predefined regions<a class="headerlink" href="#calulate-mc-across-predefined-regions" title="Permalink to this headline"></a></h1>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#methygff" id="id1">methyGff</a></p></li>
<li><p><a class="reference internal" href="#paramaters" id="id2">Paramaters</a></p></li>
<li><p><a class="reference internal" href="#output-files" id="id3">Output files</a></p></li>
<li><p><a class="reference internal" href="#output-format" id="id4">Output format</a></p></li>
</ul>
</div>
<section id="methygff">
<h2><a class="toc-backref" href="#id1">methyGff</a><a class="headerlink" href="#methygff" title="Permalink to this headline"></a></h2>
<p>Through GTF, GFF or bed files, the methylation level of the designated region and
upstream and downstream was calculated, and the methylation level matrix was generated.
The generated methylation level file and matrix file can be used to generate profile and Heatmap
visualization.</p>
<ul class="simple">
<li><p>The methratio file calculated by <span class="xref std std-doc">calmeth</span>, the format is <cite>chrom pos strand context nC nCover methlevel</cite>.</p></li>
</ul>
<p>For exsample: <cite>chr1    34      -       CHG     2       14      0.142857</cite></p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>An example usage is:
  with gtf file:
    methyGff -B -o gene.meth -G genome.fa -gtf gene.gtf -m output.methrario.txt

  with multiple gtf file:
    methyGff -B -o expressed.gene.meth unexpressed.gene.meth <span class="se">\</span>
        -G genome.fa  -gtf expressed.gene.gtf unexpressed.gene.gtf -m output.methrario.txt

  with bed file:
    methyGff -B -o gene.meth -G genome.fa -b gene.bed -m output.methrario.txt

  with multiple bed file:
    methyGff -B -o expressed.gene.meth unexpressed.gene.meth <span class="se">\</span>
        -G genome.fa -b expressed.gene.bed unexpressed.gene.bed -m output.methrario.txt
</pre></div>
</div>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>The number of input gtf/gff/bed files must be the same as the number of output prefixes..</p>
</div>
</section>
<section id="paramaters">
<h2><a class="toc-backref" href="#id2">Paramaters</a><a class="headerlink" href="#paramaters" title="Permalink to this headline"></a></h2>
<p>Command Format :   methyGff [options] -o &lt;OUT_PREFIX&gt; -G GENOME -gff &lt;GFF file&gt;/-gtf &lt;GTF file&gt;/-b &lt;bed file&gt; -m &lt;from Split methratio outfile&gt; [-B][-P]</p>
<table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head" colspan="3"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td colspan="2"><p>-o/–out</p></td>
<td><p>Output file prefix</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–genome/-G</p></td>
<td><p>Name of the genome mapped against, MUST build index first <a class="reference internal" href="bdindex.html"><span class="doc">Build index</span></a></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-m|–methratio</p></td>
<td><p>DNA methratio output file, generated by the tool <span class="xref std std-doc">calmeth</span></p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-c|–coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default:4</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-C</p></td>
<td><p>&lt;= &lt;INT&gt; coverage. default 600.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-nC</p></td>
<td><p>&gt;= &lt;INT&gt; Cs per bins or genes. default:1</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-gtf/-gff</p></td>
<td><p>Gtf/gff file</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b</p></td>
<td><p>Bed file, chrom start end</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-b4</p></td>
<td><p>Bed file, chrom start end strand</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-b5</p></td>
<td><p>Bed file, chrom start end geneid strand</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-d/–distance</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>DNA methylation level distributions in body and &lt;INT&gt;-bp flanking sequences.
The distance of upstream and downstream. default:2000</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-B/–body</p></td>
<td><p>Calculate the DNA methylation level of per region.</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-P/–promoter</p></td>
<td><p>Calculate the DNA methylation level of per region’s upstream [d]k.</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–TSS</p></td>
<td><p>Caculate matrix for TSS. [Outfile: outPrefix.TSS.cg.txt]</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–TTS</p></td>
<td><p>Caculate matrix for TTS. [Outfile: outPrefix.TTS.cg.n.txt]</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–GENE</p></td>
<td><p>Caculate matrix for TSS. [Outfile: outPrefix.TSS.cg.txt]</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–TTS</p></td>
<td><p>Caculate matrix for GENE and flank [d]k. [outPrefix.GENE.cg.txt]</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-s/–step</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>Gene body and their flanking sequences using an overlapping sliding window of 2% of
the sequence length at a step of 1% of the sequence length. So default step: 0.01 (1%)</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-bl/–bodyLen</p></td>
<td><p>Body length to which all regions will be fit. (default: same as -d)</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>-S/–chromStep</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Caculate the density of genes/TEs in chromsome using an overlapping sliding window of
100000bp at a step of 50000bp, must equal “-s” in Split.. default step: 50000(bp)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–help/-h</p></td>
<td><p>Print help</p></td>
</tr>
</tbody>
</table>
</section>
<section id="output-files">
<h2><a class="toc-backref" href="#id3">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this headline"></a></h2>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>Output</p>
</div>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. prefix.meth.AverMethylevel.txt
<span class="m">2</span>. prefix.meth.Methylevel.txt
<span class="m">3</span>. prefix.meth.TSSprofile.txt
<span class="m">4</span>. prefix.meth.centerprofile.txt
<span class="m">5</span>. prefix.col-0.meth.annoDensity.txt
<span class="m">6</span>. prefix.meth.body.c*.txt
<span class="c1"># run methyGff with -B paramater</span>
<span class="m">7</span>. prefix.bdgene.Promoter.c*.txt
<span class="c1"># run methyGff with -P paramater</span>
<span class="m">8</span>. prefix.bdgene.TSS.cg.txt
<span class="c1"># run methyGff with --TSS paramater</span>
<span class="m">9</span>. prefix.bdgene.TTS.cg.txt
<span class="c1"># run methyGff with --TTS paramater</span>
<span class="m">10</span>. prefix.bdgene.GENE.cg.txt
<span class="c1"># run methyGff with --GENE paramater</span>
</pre></div>
</div>
</section>
<section id="output-format">
<h2><a class="toc-backref" href="#id4">Output format</a><a class="headerlink" href="#output-format" title="Permalink to this headline"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="m">1</span>. AverMethylevel
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1">## N is defined by -s paramater</span>
<span class="m">2</span>. Methylevel
    CG/CHG/CHH UP/BODY/DOWN Methylevel
    <span class="c1"># per line means 1 gene/TE/region</span>
<span class="m">3</span>. TSSprofile
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1"># DNA methylation level across TSS</span>
    <span class="c1"># -d N kb, TSS upstream and downstream N kb</span>
    <span class="c1"># -s move step</span>
<span class="m">4</span>. centerprofile
    CG/CHG/CHH meth_of_bin1 bin2 bin3 ... bini binj ... binN
    <span class="c1"># DNA methylation level across region center</span>
    <span class="c1"># -d N kb, center point upstream and downstream N kb</span>
    <span class="c1"># -s move step</span>
<span class="m">5</span>. annoDensity
    chrom pos methlevel strand
    <span class="c1"># ex. Chr1    0       0.559940        +-</span>
    <span class="c1"># The density of region distributions on chromsome</span>
<span class="m">6</span>. body/Promoter
    chrom regionStart strand context C_count CT_count regionID
    <span class="c1"># ex. Chr1    3631    +       CG      45      1314    AT1G01010</span>
    <span class="c1"># This file can be used for visualization using `PlotMeth:bt2profile` or `PlotMeth:bt2heatmap`</span>
<span class="m">7</span>. TSS/TTS/GENE
    regionID meth_of_bin1 bin2 bin3 ... bini binj ... binN
    ... ...
    <span class="c1"># This file is the methylation matrix across all genes, per line represents one region (gene/TE/etc)</span>
    <span class="c1"># This file can be used for visualization using `PlotMeth:bt2heatmap`</span>
</pre></div>
</div>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BatMeth2">on github</a>.</p>
</div>
</section>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022, momocoding.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>