

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Pipeline &mdash; BatMeth2 1.1 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Alignment" href="Alignment.html" />
    <link rel="prev" title="Build index" href="bdindex.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> BatMeth2
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="example.html">Example Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="bdindex.html">Build index</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Pipeline</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#batmeth2-pipeline">BatMeth2 pipeline</a></li>
<li class="toctree-l2"><a class="reference internal" href="#batmeth2-pipeline-main-parameters">BatMeth2 pipeline main parameters</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#build-index">Build index</a></li>
<li class="toctree-l3"><a class="reference internal" href="#main-alignment-paramaters">Main Alignment paramaters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#calmeth-paramaters">Calmeth paramaters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#methygff-annoation-paramaters">MethyGff/Annoation paramaters</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#output-files">Output files</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="Alignment.html">Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="Calmeth.html">Calculate DNA methylation level</a></li>
<li class="toctree-l1"><a class="reference internal" href="MethyGff.html">Calulate mC across predefined regions</a></li>
<li class="toctree-l1"><a class="reference internal" href="bt2bigwig.html">Meth2BigWig</a></li>
<li class="toctree-l1"><a class="reference internal" href="PlotMeth.html">PlotMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">DiffMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">BatMeth2</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Pipeline</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/function/Pipeline.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="pipeline">
<h1>Pipeline<a class="headerlink" href="#pipeline" title="Permalink to this headline">¶</a></h1>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>BatMeth2 can perform one click analysis or the following modules step by step:</p>
<ul class="simple">
<li><p><a class="reference internal" href="Alignment.html"><span class="doc">Alignment</span></a></p></li>
<li><p><a class="reference internal" href="Calmeth.html"><span class="doc">Calculate DNA methylation level</span></a></p></li>
<li><p><a class="reference internal" href="MethyGff.html"><span class="doc">Calulate mC across predefined regions</span></a></p></li>
<li><p><a class="reference internal" href="bt2bigwig.html"><span class="doc">Meth2BigWig</span></a></p></li>
<li><p><a class="reference internal" href="PlotMeth.html"><span class="doc">PlotMeth</span></a></p></li>
<li><p><a class="reference internal" href="DiffMeth.html"><span class="doc">DiffMeth</span></a></p></li>
</ul>
</div>
<div class="contents local topic" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#batmeth2-pipeline" id="id1">BatMeth2 pipeline</a></p></li>
<li><p><a class="reference internal" href="#batmeth2-pipeline-main-parameters" id="id2">BatMeth2 pipeline main parameters</a></p>
<ul>
<li><p><a class="reference internal" href="#build-index" id="id3">Build index</a></p></li>
<li><p><a class="reference internal" href="#main-alignment-paramaters" id="id4">Main Alignment paramaters</a></p></li>
<li><p><a class="reference internal" href="#calmeth-paramaters" id="id5">Calmeth paramaters</a></p></li>
<li><p><a class="reference internal" href="#methygff-annoation-paramaters" id="id6">MethyGff/Annoation paramaters</a></p></li>
</ul>
</li>
<li><p><a class="reference internal" href="#output-files" id="id7">Output files</a></p></li>
</ul>
</div>
<table class="docutils align-default">
<colgroup>
<col style="width: 15%" />
<col style="width: 22%" />
<col style="width: 20%" />
<col style="width: 43%" />
</colgroup>
<thead>
<tr class="row-odd"><th class="head"><p>tool</p></th>
<th class="head"><p>input files</p></th>
<th class="head"><p>main output file(s)</p></th>
<th class="head"><p>main application</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p><a class="reference internal" href="Alignment.html"><span class="doc">Alignment</span></a></p></td>
<td><p>Single/Paired-end fastq/gz files</p></td>
<td><p>alignment sam/bam file</p></td>
<td><p>Perform DNA methylation level calculation and SNP/ASM detection</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="Calmeth.html"><span class="doc">Calculate DNA methylation level</span></a></p></td>
<td><p>BS-seq align sorted sam/bam file</p></td>
<td><p>methratio file (loci/region)</p></td>
<td><p>Perform DNA methylation visulization on chromosome, and diff analysis</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="MethyGff.html"><span class="doc">Calulate mC across predefined regions</span></a></p></td>
<td><p>methration file from calmeth</p></td>
<td><p>methlevel file on genes/TEs etc.</p></td>
<td><p>DNA methylation profile or heatmap on genes/TEs/peak regions/etc.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="bt2bigwig.html"><span class="doc">Meth2BigWig</span></a></p></td>
<td><p>methration file from calmeth</p></td>
<td><p>bigwig files (c/cg/chg/chh)</p></td>
<td><p>Convert DNA methylation file to bigwig format.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="PlotMeth.html"><span class="doc">PlotMeth</span></a></p></td>
<td><p>methy files from calmeth/methyGff</p></td>
<td><p>methy profile/heatmap/boxplot</p></td>
<td><p>visulization of DNA methylation across samples</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="DiffMeth.html"><span class="doc">DiffMeth</span></a></p></td>
<td><p>methration file from calmeth</p></td>
<td><p>Diff methy cytosines/regions</p></td>
<td><p>Perform Differential DNA methylation analysis</p></td>
</tr>
</tbody>
</table>
<div class="section" id="batmeth2-pipeline">
<h2><a class="toc-backref" href="#id1">BatMeth2 pipeline</a><a class="headerlink" href="#batmeth2-pipeline" title="Permalink to this headline">¶</a></h2>
<p>An easy-to-use, auto-run package for DNA methylation analyses:</p>
<p>Raw reads:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>BatMeth2 pipel --fastp ~/location/to/fastp <span class="se">\</span>
-1 Raw_reads_1.fq.gz -2 Raw_read_2.fq.gz <span class="se">\</span>
-g ./batmeth2index/genome.fa <span class="se">\</span>
-o meth -p <span class="m">8</span> --gff ./gene.gff
</pre></div>
</div>
<p>Or clean reads:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>BatMeth2 pipel -1 Clean_reads_1.fq.gz -2 Clean_read_2.fq.gz <span class="se">\</span>
-g ./batmeth2index/genome.fa <span class="se">\</span>
-o meth -p <span class="m">8</span> --gff ./gene.gff
</pre></div>
</div>
<p>You can always see all available command-line options via –help:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ BatMeth2 --help
</pre></div>
</div>
<ul class="simple">
<li><p>After the program runs successfully, a series of files with ‘- o’ as prefix and DNA methylation level will be generated in the output directory. Please refer to the doc for the specific output file and format details.</p></li>
<li><p>In addition, there will be an HTML report file containing basic information and statistical results of data analysis.</p></li>
</ul>
</div>
<div class="section" id="batmeth2-pipeline-main-parameters">
<h2><a class="toc-backref" href="#id2">BatMeth2 pipeline main parameters</a><a class="headerlink" href="#batmeth2-pipeline-main-parameters" title="Permalink to this headline">¶</a></h2>
<div class="section" id="build-index">
<h3><a class="toc-backref" href="#id3">Build index</a><a class="headerlink" href="#build-index" title="Permalink to this headline">¶</a></h3>
<p>Usage:  (must run this step first)</p>
<ol class="arabic simple">
<li><p>Build index using for wgbs data</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ BatMeth2 build_index genomefile
</pre></div>
</div>
<ol class="arabic simple" start="2">
<li><p>Build index using for rrbs data</p></li>
</ol>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ BatMeth2 build_index rrbs genomefile
</pre></div>
</div>
</div>
<div class="section" id="main-alignment-paramaters">
<h3><a class="toc-backref" href="#id4">Main Alignment paramaters</a><a class="headerlink" href="#main-alignment-paramaters" title="Permalink to this headline">¶</a></h3>
<table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<tbody>
<tr class="row-odd"><td colspan="3"><p><strong>[ Fastq Quality Conreol ]</strong></p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–fastp</p></td>
<td><p>fastp program location</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>If –fastp is not defined, the input file should be clean data.</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p><strong>[ Main paramaters ]</strong></p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-o</p></td>
<td><p>Name of output file prefix</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-O</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>Output of result file to specified folder, default output to current folder (./)</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p><strong>[ Aligners paramaters ]</strong></p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-g</p></td>
<td><p>Name of the genome mapped against</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-i</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>Name of input file, if paired-end. please use -1, -2,
input files can be separated by commas. eg. -1 readA.fq.gz,readB.fq.gz -2 ..</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-1</p></td>
<td><p>Name of input file left end, if single-end. please use -i</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-2</p></td>
<td><p>Name of input file left end</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-p</p></td>
<td><p>Launch &lt;integer&gt; threads</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-n</p></td>
<td><p>maximum mismatches allowed due to seq. errors [0-1]</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p><strong>[ Select aligner ]</strong></p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>–aligner</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>BatMeth2(default), bwa-meth, bsmap, bismark2,
no (exit output_prefix.sam file, no need align again)</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>–go (When aligner not BatMeth2)</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Name of the genome, contaion index build by aligner. (bwa-meth/bismark2)</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="calmeth-paramaters">
<h3><a class="toc-backref" href="#id5">Calmeth paramaters</a><a class="headerlink" href="#calmeth-paramaters" title="Permalink to this headline">¶</a></h3>
<table class="docutils align-default">
<colgroup>
<col style="width: 5%" />
<col style="width: 16%" />
<col style="width: 79%" />
</colgroup>
<tbody>
<tr class="row-odd"><td colspan="2"><p>–Qual</p></td>
<td><p>calculate the methratio while read QulityScore &gt;= Q. default:20</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–redup</p></td>
<td><p>REMOVE_DUP, 0 or 1, default 1</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–region</p></td>
<td><p>Bins for region meth calculate , default 1000bp.</p></td>
</tr>
<tr class="row-even"><td colspan="3"><p>-f</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>for sam format outfile contain methState. [0 or 1], default: 0 (dont output this file).</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–coverage</p></td>
<td><p>&gt;= &lt;INT&gt; coverage. default: 4</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>–binCover</p></td>
<td><p>&gt;= &lt;INT&gt; nCs per region. default: 3</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>–chromstep</p></td>
<td><p>&gt;= &lt;INT&gt; nCs per region. default: 3</p></td>
</tr>
<tr class="row-odd"><td></td>
<td colspan="2"><p>Chromosome using an overlapping sliding window of 100000bp at a step of 50000bpdefault step: 50000(bp)</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="methygff-annoation-paramaters">
<h3><a class="toc-backref" href="#id6">MethyGff/Annoation paramaters</a><a class="headerlink" href="#methygff-annoation-paramaters" title="Permalink to this headline">¶</a></h3>
<table class="docutils align-default">
<colgroup>
<col style="width: 4%" />
<col style="width: 17%" />
<col style="width: 79%" />
</colgroup>
<tbody>
<tr class="row-odd"><td colspan="3"><p>–gtf/–gff/–bed/–bed4/–bed5</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>gtf / gff / bed files, bed: Chr start end; bed4: Chr start end strand; bed5: Chr start end id strand;</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>-d/–distance</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>DNA methylation level distributions in body and &lt;INT&gt;-bp flanking sequences. The distance of upstream and downstream. default:2000</p></td>
</tr>
<tr class="row-odd"><td colspan="3"><p>–step</p></td>
</tr>
<tr class="row-even"><td></td>
<td colspan="2"><p>Gene body and their flanking sequences using an overlapping sliding window of 5%of the sequence length at a step of 2.5% of the sequence length. So default step: 0.025 (2.5%)</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>-C</p></td>
<td><p>&lt;= &lt;INT&gt; coverage. default:1000</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-bl/–bodyLen</p></td>
<td><p>Body length to which all regions will be fit. (default: same as -d)</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="output-files">
<h2><a class="toc-backref" href="#id7">Output files</a><a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<p>Output file format and details see “<a class="reference external" href="https://github.com/GuoliangLi-HZAU/BatMeth2/blob/master/output_details.pdf">https://github.com/GuoliangLi-HZAU/BatMeth2/blob/master/output_details.pdf</a>”.&lt;br&gt;</p>
<p>Output report details see “<a class="reference external" href="https://www.dna-asmdb.com/download/batmeth2.html">https://www.dna-asmdb.com/download/batmeth2.html</a>” .&lt;br&gt;</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BatMeth2">on github</a>.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="Alignment.html" class="btn btn-neutral float-right" title="Alignment" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="bdindex.html" class="btn btn-neutral float-left" title="Build index" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, momocoding.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>